lib/http_aws_sigv4.c

```diff
@@ -310,6 +312,7 @@ CURLcode Curl_output_aws_sigv4(struct Curl_easy *data, bool proxy)
   char provider1[MAX_SIGV4_LEN + 1]="";
   char region[MAX_SIGV4_LEN + 1]="";
   char service[MAX_SIGV4_LEN + 1]="";
+  bool sign_as_s3 = false;
```

Comment:
`s/sign_as_s3/default_unsigned_payload/g`
because sign_as_s3 seems to imply that s3 require unsigned_payload, so `default_unsigned_payload` may be more explicit. (plus if another API would require unsigned_payload by default the code will be  more straightforward to read if the boolean doesn't imply it's use only for 3s)